src/shogun2-core/src/main/java/de/terrestris/shogun2/service/PluginService.java

```diff
@@ -85,6 +95,47 @@ public String getPluginSource(String simpleClassName) throws Exception {
 	}
 
 	/**
+	 * Removes the passed plugin from all applications and afterwards deletes the plugin itself. This overrides the
+	 * generic method to delete {@link AbstractCrudService#delete(de.terrestris.shogun2.model.PersistentObject)}.
+	 *
+	 * @param plugin
+	 */
+	@Override
+	@PreAuthorize("hasRole(@configHolder.getSuperAdminRoleName()) or hasPermission(#plugin, 'DELETE')")
+	public void delete(E plugin) {
+		ApplicationService<Application, ApplicationDao<Application>> appService = this.applicationService;
+		if (appService == null) {
+			LOG.error("Plugin cannot be deleted, failed to autowire application service");
+			return;
+		}
+		// TODO We should have a more elegant way of finding the affected applications instead of fetching them all
+		//      Maybe a generic method `service.findAllHavingSubentity(String nameOfCollPropToCheck, Class subtype)`
+		//      in the AbstractCrudService?
+		List<Application> apps = appService.findAll();
+		Integer pluginId = plugin.getId();
+		for (Application app : apps) {
+			List<Plugin> plugins = app.getPlugins();
+			if (plugins != null && plugins.contains(plugin)) {
+				if (LOG.isDebugEnabled()) {
```

Comment:
I think this check is only blowing up the code...
Why not just use `LOG.debug(...)`!?